pkg/test/runner/config.go

```diff
@@ -23,13 +23,23 @@ import (
 	"gopkg.in/yaml.v3"
 )
 
+// ImperativeTestCaseConfig contains the config only for imperative
+// function run
+type ImperativeTestCaseConfig struct {
+	ExecPath string            `json:"execPath,omitempty" yaml:"execPath,omitempty"`
+	Image    string            `json:"image,omitempty" yaml:"image,omitempty"`
+	Args     map[string]string `json:"args,omitempty" yaml:"args,omitempty"`
+}
+
 // TestCaseConfig contains the config information for the test case
 type TestCaseConfig struct {
-	ExitCode int  `json:"exitCode,omitempty" yaml:"exitCode,omitempty"`
-	Network  bool `json:"network,omitempty" yaml:"network,omitempty"`
-	RunCount int  `json:"runCount,omitempty" yaml:"runCount,omitempty"`
-	Skip     bool `json:"skip,omitempty" yaml:"skip,omitempty"`
-	Debug    bool `json:"debug,omitempty" yaml:"debug,omitempty"`
+	ExitCode         int                      `json:"exitCode,omitempty" yaml:"exitCode,omitempty"`
+	Network          bool                     `json:"network,omitempty" yaml:"network,omitempty"`
+	RunCount         int                      `json:"runCount,omitempty" yaml:"runCount,omitempty"`
+	Skip             bool                     `json:"skip,omitempty" yaml:"skip,omitempty"`
+	Debug            bool                     `json:"debug,omitempty" yaml:"debug,omitempty"`
+	TestType         string                   `json:"testType,omitempty" yaml:"testType,omitempty"`
```

Comment:
What's the semantic of this? If it's eval, do you still run render on a package that has a pipeline?

Reply:
No. If it's `eval` runner will only run the eval function. If it's `render` it will only run pipeline.